template.php

```diff
@@ -1184,9 +1184,14 @@ function ec_resp_link($variables) {
     }
   }
   $path = ($variables['path'] == '<nolink>') ? '#' : check_plain(url($variables['path'], $variables['options']));
+
+  $variables += array('options' => array());
```

Comment:
I am a bit confused with the solution presented, I may be wrong, but what I am understanding is that:
We are appending an empty array always even when is not needed.

If it is what's happening I would suggest we check when the var is not set and not null and set it rather than set it in all cases.

Reply:
What does the += is adding the options array; and inside it, adding the attributes array if it does not exist.
Actually, it's quite rare that these arrays don't exist.
The option you suggest, would be to have a condition checking if options exists, and if it does, adding the attributes array; and also adding a condition when building the output.
Imo this is more complex, so i validated that patch, but maybe we should indeed ask the validation from the technical lead ?

Reply:
Hello @aritomelo,

This is needed by the default's Drupal's implementation.
Give me 5 minutes, I will copy here the link where you can see that.

Reply:
Check the default's Drupal's implementation of `theme_link()`, it needs the 'options' member: http://cgit.drupalcode.org/drupal/tree/includes/theme.inc?h=7.x#n1738

Reply:
Oh My... it's not even using `theme_link()`...

Reply:
But that doesn't change anything, see what @Fefaine said, it's perfectly valid.

Reply:
OK, th PR is ready to go.
Thx all for the input and clarification.